### ISSUE TITLE
关于Lpro损失

### ISSUE THREAD

**Issue by sunxia233** [wangguanan/light-reid#16]
感谢您的工作，给我非常大的启发。但是我刚入门对一些知识还没有想明白，比如说您在论文中的Lpro distillation loss，
zk+1和zk应该是属于不同的长度的（k更长，k+1），为什么它可以用交叉熵损失来计算呢，一般情况下，比如我们的特征和最后的标签采用交叉熵损失计算一般都需要保持维度的相同，Lsim也是同样的道理不知道这里为什么可以直接计算，不是很明白，我看了您的代码应该在这部分
loss = torch.sum(prob2 * torch.log(mini + prob2 / (prob1 + mini)), 1) + \
                   torch.sum(prob1 * torch.log(mini + prob1 / (prob2 + mini)), 1)
为啥可以直接相除，期待您的回复^^

**Comment by sunxia233** [comment]
第二，您的公式2，使用batch normalization 将真值特征化进行一个转化，我所了解的是batch normalization 将特征拉到了均值为0，方差为1 的标准正态分布，而符号函数将小于0的数返回-1，不完全是一个二进制编码了，这部分是有特殊的处理吗

**Comment by wangguanan** [comment]
你好。
Lpro distillation loss 使用 classifier 输出的 logits，不管什么的 feature，输出的 logits dim都是一样的，所以可以直接计算。
Lsim 首先计算一个 batch 内，两两feature的相似度，无论 feature dim 是多少，输出都是 bs*bs 的矩阵，因此也可以直接计算。
